### PR TITLE
MULE-17950: For multi level metadata, same values are retrieved even …

### PR DESCRIPTION
**Pull request by gsfernandes**
…when changing key parts on SOAP connectors

### CODE REVIEW

File: modules/extensions-soap-support/src/main/java/org/mule/runtime/module/extension/soap/internal/loader/SoapInvokeOperationDeclarer.java [hunk 1]
```diff
@@ -108,17 +113,39 @@ void declare(ConfigurationDeclarer configDeclarer, ClassTypeLoader loader, Set<E
...
+                                                                                inputResolver,
+                                                                                outputResolverSupplier,
                                                                                 NullMetadataResolver::new);
     operation.withModelProperty(new MetadataResolverFactoryModelProperty(() -> factory));
     operation.withModelProperty(new MetadataKeyIdModelProperty(loader.load(WebServiceTypeKey.class), KEYS_GROUP,
                                                                SOAP_INVOKE_METADATA_CATEGORY));
+
+    Map<String, String> inputResolversByParam = inputResolver
+        .entrySet().stream()
+        .collect(Collectors.toImmutableMap(Map.Entry::getKey,
```
Comment (lucianoRM) [inline modules/extensions-soap-support/src/main/java/org/mule/runtime/module/extension/soap/internal/loader/SoapInvokeOperationDeclarer.java]: static import

File: modules/extensions-soap-support/src/main/java/org/mule/runtime/module/extension/soap/internal/loader/SoapInvokeOperationDeclarer.java [hunk 2]
```diff
@@ -108,17 +113,36 @@ void declare(ConfigurationDeclarer configDeclarer, ClassTypeLoader loader, Set<E
...
+    TypeKeysResolver keysResolver = new InvokeKeysResolver();
+    OutputTypeResolver outputResolver = new InvokeOutputTypeResolver();
+    DefaultMetadataResolverFactory factory = new DefaultMetadataResolverFactory(() -> keysResolver,
+                                                                                inputResolver,
+                                                                                () -> outputResolver,
                                                                                 NullMetadataResolver::new);
     operation.withModelProperty(new MetadataResolverFactoryModelProperty(() -> factory));
     operation.withModelProperty(new MetadataKeyIdModelProperty(loader.load(WebServiceTypeKey.class), KEYS_GROUP,
-                                                               SOAP_INVOKE_METADATA_CATEGORY));
+                                                               keysResolver.getCategoryName()));
```
Comment (gsfernandes) [inline modules/extensions-soap-support/src/main/java/org/mule/runtime/module/extension/soap/internal/loader/SoapInvokeOperationDeclarer.java]: @alegmarra the value hardcoded here was added when the metadata cache was implemented but didn't match the getCategoryName() returned by the InvokeKeysResolver. Therefore I have also fixed that.

**Comment by gsfernandes** [comment]
--validate

**Comment by gsfernandes** [comment]
--validate

**Pull request merged**